src/crimson/osd/pg_backend.cc

```diff
@@ -1163,7 +1163,72 @@ PGBackend::omap_get_keys(
       ll_read_errorator::pass_further{}
     );
 }
+static
+PGBackend::omap_cmp_ertr::future<> do_omap_val_cmp(
+  std::map<std::string, ceph::bufferlist, std::less<>> out,
+  std::map<std::string, std::pair<bufferlist, int>> assertions)
+{
+  bufferlist empty;
+  for (auto &i: assertions) {
+    auto out_entry = out.find(i.first);
+    bufferlist &bl = (out_entry != out.end()) ?
+      out_entry->second : empty;
+    switch (i.second.second) {
```

Comment:
`second.second` is basically an opcode while `second.fist` is one of the operands. Yet another unpack (structured binding) could help expressing that to readers of the code.